pyscf/tdscf/uhf.py

```diff
@@ -782,6 +1057,181 @@ def vind(xys):
 
     return vind, hdiag
 
+@lib.with_doc(rhf.TDA.__doc__)
+class TDA_SF(TDBase):
+    singlet = None
```

Comment:
* extype should be initialized in the class.
* This attribute should be add to keywords list `_keys`

Reply:
OK. I will add this to the uhf.TDA_SF class.
**
extype = getattr(__config__, 'tdscf_uhf_sf_SF-TDA_extype', 0)
collinear_samples = getattr(__config__, 'tdscf_uhf_sf_SF-TDA_collinear_samples', 200)

 _keys = {'extype','collinear_samples'}
    
def __init__(self,mf,extype=0,collinear_samples=200):
    TDBase.__init__(self,mf)
    # extype is used to determine which spin flip excitation will be calculated.
    # spin flip up: exytpe=0, spin flip down: exytpe=1.
    self.extype=extype
    # collinear_samples controls the 1d spin sample points in TDDFT/TDA.
    self.collinear_samples = collinear_samples
**